File: app/src/main/java/com/water/datasample/SdFileActivity.java
package com.water.datasample;

import android.Manifest;
import android.content.Context;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.water.datasample.helper.SdFileHelper;
import com.water.utilities.PermissionHelper;

import java.io.IOException;

public class SdFileActivity extends AppCompatActivity
        implements View.OnClickListener{
    private EditText etSdFileName;
    private EditText etSdContent;
    private Button btnSdSave;
    private Button btnSdClean;
    private Button btnSdRead;
    private Context _ctx;

    private String[] _permission = {
            Manifest.permission.READ_EXTERNAL_STORAGE,
            Manifest.permission.WRITE_EXTERNAL_STORAGE
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sd_file);
        bindViews();
    }

    private void bindViews() {
        _ctx = SdFileActivity.this;

        etSdFileName = (EditText) findViewById(R.id.etSdFileName);
        etSdContent = (EditText) findViewById(R.id.etSdContent);
        btnSdSave = (Button) findViewById(R.id.btnSdSave);
        btnSdClean = (Button) findViewById(R.id.btnSdClean);
        btnSdRead = (Button) findViewById(R.id.btnSdRead);

        btnSdSave.setOnClickListener(this);
        btnSdClean.setOnClickListener(this);
        btnSdRead.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.btnSdClean:
                etSdContent.setText("");
                etSdFileName.setText("");
                break;
            case R.id.btnSdSave:
                String sfn = etSdFileName.getText().toString();
                String scontent = etSdContent.getText().toString();
                SdFileHelper sfhSave = new SdFileHelper(this);
                PermissionHelper.checkPermission(_ctx, _permission);

                try
                {
                    sfhSave.sava(sfn, scontent);
                    Toast.makeText(_ctx, "数据写入成功", Toast.LENGTH_SHORT).show();
                }
                catch(Exception e){
                    e.printStackTrace();
                    Toast.makeText(_ctx, "数据写入失败", Toast.LENGTH_SHORT).show();
                }
                break;
            case R.id.btnSdRead:
                String rcontent = "";
                SdFileHelper sfhread = new SdFileHelper(this);
                PermissionHelper.checkPermission(_ctx, _permission);

                try
                {
                    String rfn = etSdFileName.getText().toString();
                    rcontent = sfhread.read(rfn);
                }
                catch(IOException e){
                    e.printStackTrace();
                }

                Toast.makeText(_ctx, rcontent, Toast.LENGTH_SHORT).show();
                break;
        }
    }
}

File: app/src/main/java/com/github/androidhappyclub/datasample/SdFileActivity.kt
/*
 * MIT License
 *
 * Copyright (c) 2024 AndroidHappyClub
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.github.androidhappyclub.datasample

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.ave.vastgui.tools.view.toast.SimpleToast
import com.ave.vastgui.tools.viewbinding.viewBinding
import com.github.androidhappyclub.datasample.databinding.ActivitySdFileBinding
import com.github.androidhappyclub.datasample.helper.SdFileHelper
import java.io.IOException

class SdFileActivity : AppCompatActivity(R.layout.activity_sd_file) {

    private val binding by viewBinding(ActivitySdFileBinding::bind)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding.btnSdClean.setOnClickListener {
            binding.etSdContent.setText("")
            binding.etSdFileName.setText("")
        }

        binding.btnSdSave.setOnClickListener {
            val sfn = binding.etSdFileName.text.toString()
            val sContent = binding.etSdContent.text.toString()
            val sfhSave = SdFileHelper()
            if (sfhSave.sava(sfn, sContent).isSuccess) {
                SimpleToast.showShortMsg("数据写入成功")
            } else {
                SimpleToast.showShortMsg("数据写入失败")
            }
        }

        binding.btnSdRead.setOnClickListener {
            var rContent = ""
            val helper = SdFileHelper()
            try {
                val rfn = binding.etSdFileName.text.toString()
                rContent = helper.read(rfn)
            } catch (e: IOException) {
                e.printStackTrace()
            }
            SimpleToast.showShortMsg(rContent)
        }
    }

}